tests/test_frame.py

```diff
@@ -655,21 +660,23 @@ def test_types_groupby_agg() -> None:
         assert_type(df.groupby("col1").agg(["min", "max"]), pd.DataFrame), pd.DataFrame
     )
     check(assert_type(df.groupby("col1").agg([min, max]), pd.DataFrame), pd.DataFrame)
+    agg_dict1: dict[Hashable, str] = {"col2": "min", "col3": "max", 0: "sum"}
+    check(assert_type(df.groupby("col1").agg(agg_dict1), pd.DataFrame), pd.DataFrame)
+    agg_dict2: dict[Hashable, AggFuncTypeBase] = {"col2": min, "col3": max, 0: min}
+    check(assert_type(df.groupby("col1").agg(agg_dict2), pd.DataFrame), pd.DataFrame)
+    agg_dict3: dict[Hashable, str | AggFuncTypeBase] = {
+        "col2": min,
+        "col3": "max",
+        0: lambda x: x.min(),
```

Comment:
`lambda` problems are always a problem because type checkers can't infer them.  If `Callable` works, then it's OK

Reply:
So of the various dict examples, only one actually needed to be annotated, and I used a union and a `Callable` as per your suggestion. The rest, the type checkers were able to successfully choose a default type that was compatible. I'm not sure exactly what the threshold is in terms of number of variables where MyPy chooses to graduate from an inferred `Union` to an inferred `object`, here it appeared to be 3 different types, but elsewhere I've seen more.